lib/order.rb

```diff
@@ -0,0 +1,56 @@
+require_relative "customer"
+
+def string_to_hash(string)
+  hash = {}
+  array_of_kvpairs = string.split(";")
+  array_of_kvpairs.each do |pair|
+    pair_array = pair.split(":")
+    hash[pair_array[0]] = pair_array[1].to_f
+  end
+  return hash
+end
+
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+  attr_writer :products, :customer, :fulfillment_status
```

Comment:
Using `attr_accessor` for the ones in both reader and writer makes things a bit easier to read!